### PR TITLE
external-dns: release 1.2 branches

### PR DESCRIPTION
**Pull request by alebedev87**
The PR sets up the CI on the branches created for the recently shipped `release-1.2`.

### CODE REVIEW

**Comment by alebedev87** [comment]
/pj-rehearse ack

No need to waste resources for another rehearsal run because the mirroring config doesn't impact the tests in any way.

File: ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    make test
+  container:
+    from: src
+- as: e2e-aws-operator
+  steps:
+    cluster_profile: aws
+    dependencies:
+      OO_INDEX: ci-index-external-dns-operator-bundle
+    env:
+      OO_CHANNEL: stable-v1
```
Comment (gcs278) [inline ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml:53]: Why isn't this `stable-v1.2`? Doesn't this fetch the latest package from the `stable-v1` channel? Currently, it's set to 1.2?, but if we upgrade to 1.3, wouldn't this 1.2 job incorrectly use version 1.3?

**Review by gcs278 — COMMENTED** [review]
One question, otherwise looks good

File: ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml
Comment (gcs278) [inline ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml:1]: Should we update `ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-main.yaml` with the configuration removal of the `base` image replacement function? Seems odd that main would get behind this release, or are you planning to do that seperately?

Comment (gcs278) [reply ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml:1]: Sorry, operator main file is in sync. I meant on the operand, such as this:
`ci-operator/config/openshift/external-dns/openshift-external-dns-master.yaml`

File: ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml (hunk 1 above)
Comment (alebedev87) [reply ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml:53]: You are right, we should use the minor channel here, good catch!

File: ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml
Comment (alebedev87) [reply ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml:1]: Right, I submitted a separate PR for this: https://github.com/openshift/release/pull/46164.

File: ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        requests:
+          cpu: 100m
+    workflow: optional-operators-ci-gcp
+- as: e2e-azure-operator
+  steps:
+    cluster_profile: azure4
+    dependencies:
+      OO_INDEX: ci-index-external-dns-operator-bundle
+    env:
+      OO_CHANNEL: stable-v1
```
Comment (gcs278) [inline ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml:91]: I'd imagine the minor version should apply here (multiple files, and https://github.com/openshift/release/pull/46137

Comment (alebedev87) [reply ci-operator/config/openshift/external-dns-operator/openshift-external-dns-operator-release-1.2.yaml]: Right, my bad.

**Comment by gcs278** [comment]
@alebedev87 planning to rehearse? I think we changed the tests in the last round.

**Pull request merged**